ai-service/src/ai_service/errors.py

```
@@ -38,3 +38,14 @@ def empty_embedding(cls) -> "InvalidParam":
     @classmethod
     def invalid_results_count(cls) -> "InvalidParam":
         return cls("number_of_results must be an integer between 1 and 100")
+
+    @classmethod
+    def invalid_repo_url(cls) -> "InvalidParam":
+        return cls("Invalid GitHub repository URL format")
+
+    @classmethod
+    def git_clone_failed(cls, error=None) -> "InvalidParam":
```

Ruff (0.12.2)

Missing type annotation for function argument `error`

(ANN001)